Simplified/Reader2/NYPLReaderExtensions.swift

```diff
@@ -46,20 +46,36 @@ extension NYPLBook {
 
 @objc extension NYPLRootTabBarController {
   func presentBook(_ book: NYPLBook) {
-    guard let libModule = r2Owner?.libraryModule else {
+    guard let libraryService = r2Owner?.libraryService, let readerModule = r2Owner?.readerModule else {
       return
     }
 
-    let libService = libModule.libraryService
-
-    libService.parseBookAsync(book) { [weak self] publication in
+    libraryService.openBook(book, sender: self) { [weak self] result in
       guard let navVC = self?.selectedViewController as? UINavigationController else {
         preconditionFailure("No navigation controller, unable to present reader")
       }
-
-      self?.r2Owner.readerModule.presentPublication(publication,
-                                                    book: book,
-                                                    in: navVC) {}
+      switch result {
+      case .success(let publication):
+        readerModule.presentPublication(publication: publication, book: book, in: navVC) {
+          //
+        }
+      case .cancelled:
+        // .cancelled is returned when publication has restricted access to its resources and can't be rendered
+        NYPLErrorLogger.logError(nil, summary: "Error accessing book resources", metadata: [
+          "book": book
```

Comment:
`book` by itself doesn't print its details, but you can log `book.loggableDictionary` instead. Also below at line 74.

Reply:
fixed